File: src/main/java/sortingalgorithms/countingsort/java/CountingSort.java
package sortingalgorithms.countingsort.java;
import org.jetbrains.annotations.NotNull;

public class CountingSort {
    public void sort(@NotNull Integer[] input, int max) {
        int[] countArray = new int[max+1];
        for (int num : input) {
            countArray[num]++;
        }

        int k = 0;
        int countArrayLength = countArray.length;
        for (int i = 0; i < countArrayLength; i++) {
            for (int j = 0; j < countArray[i]; j++) {
                input[k++] = i;
            }
        }
    }
}

File: src/main/java/sortingalgorithms/countingsort/kotlin/CountingSort.kt
package sortingalgorithms.countingsort.kotlin

class CountingSort {

    fun sort(inputData: IntArray, max: Int) {
        var countingArray = Array(max + 1) {0}
        for (number in inputData) {
            countingArray[number]++
        }

        var k = 0
        for (i in countingArray.indices) {
            for (j in 0 until countingArray[i]) {
                inputData[k++] = i
            }
        }
    }
}